apps/dotcom/src/components/DocumentName/DocumentName.tsx

```diff
@@ -0,0 +1,295 @@
+import {
+	Button,
+	DropdownMenu,
+	Kbd,
+	OfflineIndicator,
+	TLDocument,
+	track,
+	useActions,
+	useBreakpoint,
+	useEditor,
+	useTranslation,
+} from '@tldraw/tldraw'
+import {
+	ChangeEvent,
+	KeyboardEvent,
+	ReactNode,
+	SetStateAction,
+	useCallback,
+	useEffect,
+	useLayoutEffect,
+	useRef,
+	useState,
+} from 'react'
+import { FORK_PROJECT_ACTION } from '../../utils/sharing'
+import { SAVE_FILE_COPY_ACTION } from '../../utils/useFileSystem'
+import { getShareUrl } from '../ShareMenu'
+
+type NameState = {
+	readonly name: string | null
+	readonly isEditing: boolean
+}
+
+const MAX_TITLE_WIDTH_PX = 420
+
+export const DocumentTopZone = track(function DocumentTopZone({
+	isOffline,
+}: {
+	isOffline: boolean
+}) {
+	const isDocumentNameVisible = useBreakpoint() >= 4
+
+	return (
+		<DocumentTopZoneContainer>
```

Comment:
philosophical q: why have this be separated from the `tlui-menu-zone`? why not have it next to the Page menu area?

Reply:
it might end up making this a bit simpler b/c then it wouldn't duplicate the other actions (save & fork)